wlib/stl/List.cpp

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file List.cpp
+ * @brief List is an implementation of the stl doubly-linked link
+ *
+ * It provides functionality for pushing popping from front and back
+ * and search and indexing
+ *
+ * @author Ambareesh Balaji
+ * @date November 4, 2017
+ * @bug No Known bugs
+ */
+
+#include "../memory/Memory.h"
+#include "List.h"
+
+using namespace wlp;
+// Constructor
+template<class T>
+List<T>::List() :
```

Comment:
Please rename this class `LinkedList` or something. We might want to use `List` as a parent class if we do other implementations in the future.

Reply:
How about DList to imply it's doubly linked?

Reply:
I guess `DLList` would do as 'Doubly Linked List'